internal/pkg/otel/manager/execution_subprocess.go

```diff
@@ -212,6 +221,35 @@ func (r *subprocessExecution) startCollector(ctx context.Context, logger *logger
 	return ctl, nil
 }
 
+func (r *subprocessExecution) getCollectorPorts() (healthCheckPort int, metricsPort int, err error) {
```

Comment:
Nit: would this be a simpler / easier to follow implementation?


```suggestion
func (r *subprocessExecution) getCollectorPorts() (healthCheckPort int, metricsPort int, err error) {
	if r.collectorHealthCheckPort > 0 {
		healthCheckPort = r.collectorHealthCheckPort
	} else {
		ports, findPortsErr := findRandomTCPPorts(1)
		if findPortsErr != nil {
			return 0, 0, err
		}
		healthCheckPort = ports[0]
	}

	if r.collectorMetricsPort > 0 {
		metricsPort = r.collectorMetricsPort
	} else {
		ports, findPortsErr := findRandomTCPPorts(1)
		if findPortsErr != nil {
			return 0, 0, err
		}
		metricsPort = ports[0]
	}

	return healthCheckPort, metricsPort, nil
}
```

It does require calling `findRandomTCPPorts` twice.  Maybe you were trying to avoid that for some reason?